gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/SpecimenMapperTest.java

```diff
@@ -0,0 +1,117 @@
+
+    @Mock
+    private DateTimeMapper dateTimeMapper;
+
+    @InjectMocks
+    private SpecimenMapper specimenMapper;
+
+    @Test
+    public void testSpecimenIsMapped() {
+        when(dateTimeMapper.mapDateTime(any())).thenCallRealMethod();
+        RCMRMT030101UK04EhrExtract ehrExtract = unmarshallEhrExtract("specimen_valid.xml");
+        List<Specimen> specimenList = specimenMapper.mapSpecimen(
+            ehrExtract, List.of(DIAGNOSTIC_REPORT_WITH_SPECIMEN), PATIENT, PRACTICE_CODE
+        );
+
+        assertThat(specimenList).isNotEmpty();
+
+        final Specimen specimen = specimenList.get(0);
+        checkFixedValues(specimen);
+        assertThat(specimen.getNote().get(0).getText()).isEqualTo(NOTE_TEXT);
+        assertThat(specimen.getAccessionIdentifier().getValue()).isEqualTo(ACCESSION_IDENTIFIER_VALUE);
+        assertThat(specimen.getType().getText()).isEqualTo(SPECIMEN_TYPE_TEXT);
+        assertThat(specimen.getCollection().getCollected().toString()).isEqualTo(SPECIMEN_COLLECTED_DATETIME.toString());
+    }
+
+    @Test
+    public void testSpecimenIsMappedWithNoOptionalFields() {
+        RCMRMT030101UK04EhrExtract ehrExtract = unmarshallEhrExtract("specimen_no_optional_fields.xml");
+        List<Specimen> specimenList = specimenMapper.mapSpecimen(
+            ehrExtract, List.of(DIAGNOSTIC_REPORT_WITH_SPECIMEN), PATIENT, PRACTICE_CODE
+        );
+
+        Specimen specimen = specimenList.get(0);
+        assertThat(specimen.getCollection().getCollected()).isNull();
+        assertThat(specimen.hasAccessionIdentifier()).isFalse();
+        assertThat(specimen.getType().getText()).isNullOrEmpty();
+        checkFixedValues(specimen);
+    }
+
+    private void checkFixedValues(Specimen specimen) {
+        assertThat(specimen.getId()).isEqualTo(TEST_SPECIMEN_ID);
+        assertThat(specimen.getMeta().getProfile().get(0).getValue()).isEqualTo(SPECIMEN_META_PROFILE);
+        assertThat(specimen.getIdentifierFirstRep().getSystem()).isEqualTo(IDENTIFIER_SYSTEM);
+        assertThat(specimen.getIdentifierFirstRep().getValue()).isEqualTo(TEST_SPECIMEN_ID);
+        assertThat(specimen.getSubject().getResource().getIdElement().getValue()).isEqualTo(PATIENT.getId());
+    }
+
+    @Test
+    public void testInvalidSpecimenIsNotMapped() {
+        RCMRMT030101UK04EhrExtract ehrExtract = unmarshallEhrExtract("specimen_invalid.xml");
+        List<Specimen> specimenList = specimenMapper.mapSpecimen(
+            ehrExtract, List.of(DIAGNOSTIC_REPORT_WITHOUT_SPECIMEN), PATIENT, PRACTICE_CODE
+        );
+
+        assertThat(specimenList).isEmpty();
+    }
+
+    private static DiagnosticReport generateDiagnosticReportWithSpecimenReference() {
+        DiagnosticReport diagnosticReport = (DiagnosticReport) new DiagnosticReport().setId("DIAGNOSTIC_REPORT_ID");
+        List<Reference> specimen = List.of(new Reference(SPECIMEN_PREFIX + TEST_SPECIMEN_ID));
```

Comment:
"DIAGNOSTIC_REPORT_ID" could be set as variable, as it's used here and in line 109